modes/python-mode/lem-python-mode.asd

```diff
@@ -1,7 +1,18 @@
 (defsystem "lem-python-mode"
-  :depends-on ("lem-core"
-               #+#.(cl:if (ql:where-is-system :async-process) '(and) '(or)) "lem-process")
+  :depends-on ("lem-core")
+  :defsystem-depends-on ("uiop")
   :serial t
-  :components ((:file "python-mode")
-               #+#.(cl:if (ql:where-is-system :async-process) '(and) '(or))
-               (:file "run-python")))
+  :components ((:file "python-mode")))
+
+;;; This is the recommended way to include optional dependencies.
+;;; See: https://common-lisp.net/project/asdf/asdf/The-defsystem-grammar.html#index-_003aweakly_002ddepends_002don
+
+(defsystem "lem-python-mode/run"
+  :depends-on ("lem-pyton-mode" "lem-process")
+  :serial t
+  :components ((:file "run-python")))
+
+(defmethod perform :after (operation
+                           (system (eql (find-system "lem-python-mode"))))
+  (when (uiop:symbol-call :quicklisp :where-is-system :async-process)
```

Comment:
This is going to blow up on systems that do not have Quicklisp installed (e.g. ASDF-only systems). Ensure that the QL package can be found first before executing this code.